src/loaders/lottie/tvgLottieLoader.cpp

```diff
@@ -345,7 +345,19 @@ bool LottieLoader::frame(float no)
 
 float LottieLoader::totalFrame()
 {
-    return frameCnt;
+    if(selectedSection == LottieMarker::SelectedSegment::NONE) return frameCnt;
+    if(!comp) sync();
+    if(selectedSection == LottieMarker::SelectedSegment::MARKER)
+    {
+        for(auto it : comp->markers)
+            if(std::strcmp(selectedMarker.c_str(), it->name))
+                return it->time;
+    }
+    else if(selectedSection == LottieMarker::SelectedSegment::RANGE)
+    {
+        return frameCnt * selectedSegmentBegin;
+    }
+    return -1.0f;
```

Comment:
For enumeration condition, rather than `if-else`, `switch-case` would be readable.